policies/coding-style.md

```diff
@@ -584,28 +594,93 @@ less complicated than coding pure assembly, and it doesn't provide the
 same performance guarantee across different micro-architecture. Nor is
 it portable enough to meet our multi-platform support goals.
 
-## Chapter 14:  Portability
+## Chapter 14: Portability
 
 To maximise portability the version of C defined in ISO/IEC 9899:1990
 should be used. This is more commonly referred to as C90. ISO/IEC 9899:1999
 (also known as C99) is not supported on some platforms that OpenSSL is
 used on and therefore should be avoided.
 
-## Chapter 15: Miscellaneous
+## Chapter 15: Expressions
 
-Do not use `!` to check if a pointer is NULL, or to see if a str...cmp
-function found a match.  For example, these are wrong:
+Avoid needless parentheses as far as reasonable.
+For example, do not write
+```c
+    if ((p == NULL) && (!f(((2 * x) + y) == (z++))))
+```
+but
+```c
+    if (p == NULL && !f(2 * x + y == z++)).
+```
 
+For clarity, always put parentheses when mixing `&&` and `||` operations,
+comparison operators like `<=` and `==`, and bitwise operators.
```

Comment:
Looking at the sentence again, it wouldn't be very hard to make it clearer:

```suggestion
For clarity, always put parentheses when mixing operators of the same category,
such as `&&` and `||` operations, or comparison operators like `<=` and `==`, 
or bitwise operators.
```

Reply:
Hmm, when re-reading the above suggestion, which does include an improvement, I do not like the "category" and "such as" very much.
For instance, this could also be understood that one should put parentheses when mixing `*` and `+`, which is not intended.
So I propose a further refinement as follows:

For clarity, always put parentheses when mixing the Boolean `&&` and `||` operations,
when mixing comparison operators like `<=` and `==`, and when when mixing bitwise operators.

Reply:
> For clarity, always put parentheses when mixing the Boolean `&&` and `||` operations, when mixing comparison operators like `<=` and `==`, and when when mixing bitwise operators.

"Boolean" should be "Logical", to match C parlance.
The last "and" should probably be "or"
Drop the double "when"

With those mods, I approve

Reply:
Oh, and "operations" should be "operators"

Reply:
Yes, and I suggest including in the sentence also examples for the bitwise operators.
So:

For clarity, always put parentheses when mixing the logical `&&` and `||` operators,
mixing comparison operators like `<=` and `==,` or mixing bitwise operators like  `&` and `|` .

Reply:
Any changes will require a new vote -- too many have voted to change the wording now.
We don't have a process to abort a vote without it failing.

Reply:
> Any changes will require a new vote -- too many have voted to change the wording now. We don't have a process to abort a vote without it failing.

Does that mean that a new PR is needed, or can we cast a new vote in the same PR?
After all, our votes are tied to a specific *commit*, is it not?

Reply:
In my view a new PR would be needlessly inefficient.
How about doing the above clarification and the one I proposed in https://github.com/openssl/technical-policies/pull/39#discussion_r864516234
and then restart the voting on the new state?

(As apparently needed according to the policy, the current vote will need to be made fail first.)

Reply:
@DDvO I agree with Richard: IMHO a PR which has been voted down should be closed. In cases like this one, were it just failed because changes were requested, it can immediately be replaced by a new PR from the same branch (or with some suffix like `coding-style-expressions-r2`, as you prefer).

Having multiple votes within the same PR would make the thread history too confusing. But at this moment, that's just my personal opinion Your thought, @openssl/otc ?

Reply:
Make it an **or**, not an **and**

> For clarity, always put parentheses when mixing the Boolean && and || operations,
> when mixing comparison operators like <= and ==, **or** when when mixing bitwise operators.

Reply:
> Any changes will require a new vote -- too many have voted to change the wording now. We don't have a process to abort a vote without it failing.

I don't think there is a need for different failure modes. If the vote fails, it fails. But if the OTC members indicate that a revised version would be accepted, then discussion can be immediately restarted on a new PR.

Reply:
> @DDvO I agree with Richard: IMHO a PR which has been voted down should be closed. In cases like this one, were it just failed because changes were requested, it can immediately be replaced by a new PR from the same branch (or with some suffix like `coding-style-expressions-r2`, as you prefer).

True, at least when using the same branch, a new PR is very easily set up.
Yet a pointer to the previous PR should be given such that original discussion can be found quickly.

> Having multiple votes within the same PR would make the thread history too confusing. But at this moment, that's just my personal opinion Your thoughts, @openssl/otc ?

That's a good point in favor of a new PR.

Reply:
The vote includes the hash.  I don't see a big problem having a second vote here.
Another alternative would be an issue for the vote, keeping this PR open.

Reply:
The current process IMO does not allow for second vote. If anyone wants to propose a change to it no problem. I suppose we would need to change it to always use a separate issue for the votes.

Reply:
@t8m this is actually a very good idea: we shouldn't carry out the vote in the PR, but always create a dedicated Issue for a vote, with a standardized subject line (`VOTE: <subject>`). In this way, we could decouple the PR from the vote. Nevertheless, we need to amend the policies to regulate things like a shortened waiting period, if desired.